src/connector/src/error.rs

```diff
@@ -12,50 +12,13 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use thiserror::Error;
-
-#[derive(Error, Debug)]
-pub enum ConnectorError {
-    #[error("Parse error: {0}")]
-    Parse(&'static str),
-
-    #[error("Invalid parameter {name}: {reason}")]
-    InvalidParam { name: &'static str, reason: String },
-
-    #[error("Kafka error: {0}")]
-    Kafka(#[from] rdkafka::error::KafkaError),
-
-    #[error("Config error: {0}")]
-    Config(
-        #[source]
-        #[backtrace]
-        anyhow::Error,
-    ),
-
-    #[error("Connection error: {0}")]
-    Connection(
-        #[source]
-        #[backtrace]
-        anyhow::Error,
-    ),
-
-    #[error("MySQL error: {0}")]
-    MySql(#[from] mysql_async::Error),
-
-    #[error("Postgres error: {0}")]
-    Postgres(#[from] tokio_postgres::Error),
-
-    #[error("Pulsar error: {0}")]
-    Pulsar(
-        #[source]
-        #[backtrace]
-        anyhow::Error,
-    ),
```

Comment:
Does it mean that we lose some error context for mysql/postgres? (For Pulsar it seems ok)

Reply:
Can you elaborate more on the "context"?

Reply:
"Postgres error:"

Reply:
Previously, this is added at the conversion from `tokio_postgres::Error` to `ConnectorError`, but now this is not. 

TBH I think it's not very useful. But it can be useful if `tokio_postgres::Error` is very vague. In this case, `Postgres error: blabla` is more helpful.

I think ideally we should add `context()` according to the higher level action we are doing though.

Reply:
> I think ideally we should add `context()` according to the higher level action we are doing though.

Agree.

To encourage developers to provide the necessary context, a good approach would be to define `ConnectorError` as an enum and utilize `thiserror_ext::ContextInto` for conversion. 😄

This PR primarily prepares for the next one to wrap the bare `anyhow::Error` within `ConnectorError`. Once this is done, we can then consider whether switching it back to an enum implementation is appropriate.